sdk/storage/Azure.Storage.Blobs.Batching/src/Multipart.cs

```diff
@@ -0,0 +1,257 @@
+                using var body = new BufferedReadStream(section.Body, BatchConstants.ResponseLineSize);
+
+                // The first line is the status like "HTTP/1.1 202 Accepted"
+                string line = await body.ReadLineAsync(async, cancellationToken).ConfigureAwait(false);
+                string[] status = line.Split(new char[] { ' ' }, 3, StringSplitOptions.RemoveEmptyEntries);
+                if (status.Length != 3)
+                {
+                    throw BatchErrors.InvalidHttpStatusLine(line);
+                }
+                response.SetStatus(int.Parse(status[1], CultureInfo.InvariantCulture));
+                response.SetReasonPhrase(status[2]);
+
+                // Continue reading headers until we reach a blank line
+                line = await body.ReadLineAsync(async, cancellationToken).ConfigureAwait(false);
+                while (!string.IsNullOrEmpty(line))
+                {
+                    // Split the header into the name and value
+                    int splitIndex = line.IndexOf(':');
+                    if (splitIndex <= 0)
+                    {
+                        throw BatchErrors.InvalidHttpHeaderLine(line);
+                    }
+                    var name = line.Substring(0, splitIndex);
+                    var value = line.Substring(splitIndex + 1, line.Length - splitIndex - 1).Trim();
+                    response.AddHeader(name, value);
+
+                    line = await body.ReadLineAsync(async, cancellationToken).ConfigureAwait(false);
+                }
+
+                // Copy the rest of the body as the response content
+                var responseContent = new MemoryStream();
+                if (async)
+                {
+                    await body.CopyToAsync(responseContent).ConfigureAwait(false);
+                }
+                else
+                {
+                    body.CopyTo(responseContent);
+                }
+                responseContent.Seek(0, SeekOrigin.Begin);
+                response.ContentStream = responseContent;
+            }
+
+            // Collect the responses and order by Content-ID
+            Response[] ordered = new Response[responses.Count];
+            for (int i = 0; i < ordered.Length; i++)
+            {
+                ordered[i] = responses[i];
+            }
+            return ordered;
+        }
+
+        /// <summary>
+        /// Read the next line of text.
+        /// </summary>
+        /// <param name="stream">The stream to read from.</param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <returns>The next line of text.</returns>
+        internal static async Task<string> ReadLineAsync(
```

Comment:
Should this be organized under extension methods class?

Reply:
I don't think so.  It's only used directly above.